libs/two/data-access/src/lib/services/status.service.ts

```diff
@@ -0,0 +1,130 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { map, reduce } from 'rxjs/operators';
+import date from 'date-and-time';
+
+import { IChartConfiguration } from '@tamu-gisc/charts';
+import { EnvironmentService } from '@tamu-gisc/common/ngx/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class StatusService {
+  private resource: string;
+
+  constructor(private env: EnvironmentService, private http: HttpClient) {
+    this.resource = this.env.value('two_dashboard_api_url') + 'status';
+  }
+
+  public status() {
+    return this.http.get<Array<Partial<IStatusResponse>>>(this.resource);
+  }
+
+  public siteHistory(siteCode: string, history: number) {
+    const now = new Date();
+    const dateRange = [];
+    for (let i = history; i > 0; i--) {
+      const lowerDate = date.addDays(now, -1 * i);
+      const upperDate = date.addDays(now, -1 * (i - 1));
+      dateRange.push({
+        upperDate: date.format(upperDate, 'YYYY-MM-DD'),
+        lowerDate: date.format(lowerDate, 'YYYY-MM-DD')
+      });
+    }
```

Comment:
> // there's a reason this is done backwards, I just can't remember why
>    // since there's no array.mapBackwards() this will stay as is until I can remember why

I missed the iteration order. You can do `Array.reverse()` and then map them in that order but I suppose it's fine to leave as is in this case.